File: app/src/main/java/com/example/adscreen/data/Ad.kt
package com.example.adscreen.data

import com.google.firebase.firestore.DocumentId
import com.google.firebase.firestore.PropertyName

data class Ad(
    @DocumentId val id: String = "",
    @get:PropertyName("imageUrl") val imageUrl: String = "",
    @get:PropertyName("type") val type: String = "image", // 'image' or 'video'
    @get:PropertyName("duration") val duration: Long = 10,
    @get:PropertyName("assignedTablets") val assignedTablets: List<String> = emptyList()
)

File: lib/models/ad.dart
class Ad {
  final String id;
  final String type; // 'image' or 'video'
  final String url;
  final int duration; // seconds
  final List<String> assignedTablets;

  Ad({
    required this.id,
    required this.type,
    required this.url,
    required this.duration,
    required this.assignedTablets,
  });

  factory Ad.fromMap(String id, Map<String, dynamic> map) {
    return Ad(
      id: id,
      type: map['type'] ?? 'image',
      url: map['url'] ?? '',
      duration: (map['duration'] ?? 10) as int,
      assignedTablets: List<String>.from(map['assignedTablets'] ?? []),
    );
  }
}
